src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Input/Command/CommandManager.cs

```diff
@@ -352,9 +352,10 @@ internal static void TranslateInput(IInputElement targetElement, InputEventArgs
 
             // Determine UIElement/ContentElement/Neither type
             DependencyObject targetElementAsDO = targetElement as DependencyObject;
-            bool isUIElement = InputElement.IsUIElement(targetElementAsDO);
-            bool isContentElement = !isUIElement && InputElement.IsContentElement(targetElementAsDO);
-            bool isUIElement3D = !isUIElement && !isContentElement && InputElement.IsUIElement3D(targetElementAsDO);
+            // todo patternmatching: Would an as cast and null checks be faster here instead of casting multiple times?
```

Comment:
@batzen If you are suggesting, something like?
localInputBindings = ((targetElement as UIElement) ?? (targetElement as ContentElement) ?? (targetElement 
 as UIElement3D))?.InputBindingsInternal;

"is" actually casts the object. Above line will just save redundant cast.

Reply:
I was meant for the casts later inside the ifs. Getting the InputBindings here would be too early.
Maybe it's also ok as it is, as any change to the flags would make the code more difficult to read.
Just wanted to leave so todos for things like this.
If no one thinks we should have a further look at this place i can just delete the todo. ;-)

Reply:
Agreed. It won't work, unless inherited from the same interface. It will need more refactoring, than just converting that to one line.